algorithms/linfa-clustering/src/k_means/algorithm.rs

```diff
@@ -444,6 +444,31 @@ impl<F: Float, DA: Data<Elem = F>, D: Distance<F>> PredictRef<ArrayBase<DA, Ix1>
     }
 }
 
+impl<F: Float, DA: Data<Elem = F>, D: Distance<F>> PredictInto<ArrayBase<DA, Ix2>, Array1<usize>>
```

Comment:
Can you add some tests for `predict_into` just to see that it doesn't crash? You can change some of the existing tests that call `predict` to call `predict_into`.

Reply:
Of course. Before the tests, I just wanted to get some guidance on the basic structure of how this will be done.

Reply:
I've got no problems with the structure of the new trait. It's close to the structure of `PredictRef` which makes replacement all the easier.

Reply:
Since `PredictRef` now uses `PredictInto` internally, does it still make sense to make tests for `PredictInto`?